[31기 남용현] ADD: login, main 페이지 필수 및 추가 구현 완료

sunnyfterrain opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표
- [Mission1] 로그인 페이지 레이아웃
- [Mission2] id, pw 입력 시 로그인 버튼 활성화 기능
- [Mission3] 메인 페이지 레이아웃
- [Mission4] 댓글 내용 입력 후 Enter press, 혹은 게시 버튼 클릭 시 댓글 추가 기능
- [Mission5] ID, PW validation
- [Mission6] 댓글 좋아요 / 삭제 기능
- [Mission7] 아이디 검색 기능
- [Mission8] nav 프로필 사진 클릭 시 메뉴 박스 생성
- [Mission9] 반응형 구현
<br />

## :: 구현 사항 설명 
1. 로그인 페이지 개발 
- id 1글자이상 & pw 1글자 이상 일시 로그인 버튼 활성화
- 아이디 '@' & 비밀번호 '5글자' 이상이 아닐시 alert 출력 유효성 검사
- id & pw 첫글자 공백 입력시 alert 출력

2. 메인 페이지 개발
- 댓글 내용 입력 후 Enter press 및 게시 버튼 클릭 시 댓글 추가
- 댓글 좋아요 및 마우스 댓글 hover시 삭제버튼 구현
- 아이디 검색기능 추가 ( 대소문자 구문 없이 검색 가능, 한글과 영어 검색가능)
- nav 프로필 사진 클릭 시 메뉴 박스 생성 및 다른 부분 클릭 시 메뉴 박스 닫힘 구현
- 반응형 구현 (브라우저 1000px 이하시 우측 레이아웃 숨김, 640px 이하시 검색창 숨김)

<br />

## :: 성장 포인트 
- 댓글 추가 시 첫글자가 공백이면 공백 제거 기능과 공백만 입력시에도 댓글이 등록되는걸 고치기 힘들었으나,
정규화 라는것을 알게되어 겨우 해결
- 검색 기능이 조금 까다로웠는데 filter와 문자열 포함여부를 공부함에 있어서 도움이 많이 되었고,
검색 결과 창 초기화가 어려웠는데 생각보다 너무 간단해서  당황했지만 많은 도움이 되었습니다.
- 모달창 외부 닫힘에 대해서 몰랐던 기능을 알게되어서 좋았습니다. (오버레이)

<br />

## :: 기타 질문 및 특이 사항
- 데이터 로드를 fetch로 구현하였는데, html을 vscode내 라이브서버 익스텐션으로 열지 않으면 
데이터를 불러오지 못함

iamsonghee commented:
위스타그램 초기세팅

JHyeon0 commented on students/18th/songheechoi/js/login.js:
```diff
@@ -0,0 +1,46 @@
+        btnLogin.classList.add("enable");
+        btnLogin.removeAttribute("disabled");
+        return true;
+        // console.log(id,pwd);
```
PR을 올리실 때는 불필요한 주석, console.log는 삭제해 주시기 바랍니다 :)

iamsonghee replied:
네 수정완료하였습니다!

JHyeon0 commented on students/18th/songheechoi/js/login.js:
```diff
@@ -0,0 +1,46 @@
+    }
+}
+function checkValidation(id, pwd){
+    // console.log(id.indexOf("@"), pwd.length);
```
불필요한 주석 삭제 해주시면 참 좋겠다~🙃

iamsonghee replied:
네 수정완료하였습니다!

JHyeon0 commented on students/18th/songheechoi/js/login.js:
```diff
@@ -0,0 +1,46 @@
+    checkInputs(loginId.value, passWord.value);
+})
+
+document.addEventListener("keyup",function(e){
+    if(checkInputs(loginId.value, passWord.value)){
+        if(e.keyCode === 13){
+            btnLogin.click();
+        }
+    }
+})
```
아이디와 패스워드 validation을 함수로 분리하셨네요! 너무 좋습니다 :)
조건문문이 두번 중첩되어 있는데 아래와 같은 코드로 수정할 수 있을 것 같네요!
```
checkInputs(loginId.value, passWord.value) && e.keyCode ===13
```

JHyeon0 replied:
또한 form 태그를 사용하면 엔터를 눌렀는지 확인하는 로직 없이도
Enter가 form을 submit 하게 됩니다.
html파일에서 form 태그를 사용해서 한번 적용해보세요 :)

JHyeon0 commented on students/18th/songheechoi/js/main.js:
```diff
@@ -0,0 +1,55 @@
+const commentsList = document.querySelector(".feed__commentList");
+const logo = document.querySelector(".logo");
+
+// const likeButton = 
```
다음 PR에 불필요한 주석이 남아있는지 눈 크게뜨고 확인할 예정입니다!! 🙄

JHyeon0 commented on students/18th/songheechoi/main.html:
```diff
@@ -0,0 +1,169 @@
+            <button class="user-component__column btn-none">팔로우</button>
+          </div>
+        </div>
+        <div class="site-info">
+          <div>
+            <ul>
+              <li>
+                <a href="https://about.instagram.com/" target="_blank">소개</a>
+              </li>
+              <li><a href="#">도움말</a></li>
+              <li><a href="#">홍보 센터</a></li>
+              <li><a href="#">API</a></li>
+              <li><a href="#">채용 정보</a></li>
+              <li><a href="#">개인정보처리방침</a></li>
+              <li><a href="#">약관</a></li>
+              <li><a href="#">위치</a></li>
+              <li><a href="#">해시태그</a></li>
+              <li><a href="#">언어</a></li>
+            </ul>
+          </div>
+          <div>© 2021 INSTAGRAM FROM FACEBOOK</div>
+        </div>
+      </div>
```
semantic 태그 중 small 태그에 대해 검색해보시고 적용해 보시면 좋을 거 같네요!

JHyeon0 commented on students/18th/songheechoi/main.css:
```diff
@@ -0,0 +1,201 @@
+  font-weight: 600;
+}
+
+.feed__update {
+  font-size: 10px;
+  opacity: 0.3;
+  padding: 0px 5px 15px 15px;
+}
+
+.feed__myComment {
+  border-top: var(--main-border);
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
```
css 속성 순서를 지켜주세요!
제가 리뷰를 따로 남기지 않은 다른 부분도 순서! 확인해주세요 :)

[CSS property 순서]

- Layout Properties (position, float, clear, display)
- Box Model Properties (width, height, margin, padding)
- Visual Properties (color, background, border, box-shadow)
- Typography Properties (font-size, font-family, text-align, text-transform)
- Misc Properties (cursor, overflow, z-index)

JHyeon0 commented on students/18th/songheechoi/js/main.js:
```diff
@@ -0,0 +1,55 @@
+commentInput.addEventListener("input",test)
+commentInput.addEventListener("keyup", function(e){
+    // console.log(e);
+    if(e.keyCode ===13&&commentInput.value){
```
이렇게 가독성을 높이면 더 좋을거 같아요!
```suggestion
    if(e.keyCode ===13 && commentInput.value){
```

JHyeon0 commented on students/18th/songheechoi/js/main.js:
```diff
@@ -0,0 +1,55 @@
+    return comment;
+}
+
+logo.addEventListener("click",function(){
+    location.href = "./login.html";
+})
```
js에서 페이지 이동을 구현할수도 있지만
html에서 a 태그를 사용해서 구현해야 합니다.
그 이유에 대한 [링크](https://stackoverflow.com/a/1667512)입니다!

JHyeon0 commented on students/18th/songheechoi/style/main.css:
```diff
@@ -0,0 +1,221 @@
+@import "./user-component.css";
+
+nav {
+  height: 55px;
+  display: flex;
+  justify-content: space-around;
+  align-items: center;
+  padding: 12px 0px;
+  border-bottom: var(--main-border);
+  background-color: #ffffff;
+  position: fixed;
+  width: 100%;
+  top: 0;
+}
```
여기를 포함한 다른 곳에서도, CSS 순서! 꼭 확인해서 수정해주세요😀

JHyeon0 commented on students/18th/songheechoi/style/main.css:
```diff
@@ -0,0 +1,221 @@
+  /* height: 100px; */
+}
+
+.inPad {
+  padding: 10px 15px;
+}
+.small-logo {
+  font-size: 26px;
+}
+.nav__search {
+  width: 215px;
+  position: relative;
+}
+.search__bar {
```
이 곳을 포함하여 다른 곳에서도 가독성을 위해 한 줄씩 띄워주세요!

JHyeon0 commented on students/18th/songheechoi/style/main.css:
```diff
@@ -0,0 +1,221 @@
+  line-height: 18px;
+}
+
+.comment fa-heart {
+}
```
불필요한 코드는 삭제 후 PR 해주세요 :)

JHyeon0 commented on students/18th/songheechoi/login.html:
```diff
@@ -0,0 +1,33 @@
+    <title>Document</title>
+  </head>
+  <body>
+    <div class="main-box">
+      <header class="logo big-logo">westagram</header>
+      <div class="inner-box">
+        <input
+          type="text"
+          name=""
+          id="loginId"
+          placeholder="전화번호,사용자 이름 또는 이메일"
+        />
+        <input type="password" name="" id="passWord" placeholder="비밀번호" />
+        <button class="btnLogin" disabled>로그인</button>
+        <div class="line">
+          <div class="line-through"></div>
+          <div>또는</div>
+          <div class="line-through"></div>
+        </div>
+      </div>
+      <a class="forget-pwd" href="#">비밀번호를 잊으셨나요?</a>
+    </div>
```
section, form 태그를 사용하여 semantic 하게 수정 할 수 있습니다.

JHyeon0 reviewed (commented):
잘 하고 계십니다!
main태그와 article 태그를 적절히 활용하여 semantic하게 잘 해주셨네요.
추가적으로 div 대신에 section 태그를 적용할 수 있을 만한 곳이 보이네요. 확인 후 한 번 적용해보세요 :)
css 작성 순서 convention을 지키는 습관을 지금부터 잘 잡으시면 더더욱 좋을 거 같습니다.
불필요한 줄, 주석, 가독성을 조금 더 신경써서 PR을 올려주세요 👍

sunnyfterrain committed:
Add: 로그인 페이지 구현
- [Mission1] 로그인 페이지 레이아웃 구현
- [Mission2] id, pw입력 시 로그인 버튼 활성화 기능 구현

sunnyfterrain committed:
Add: 메인페이지 레이아웃
- [Mission3] 메인페이지 ui 레이아웃 제작 완료

sunnyfterrain committed:
Add: 댓글 기능 구현 및 유효성검사 구현
- [Mission4] 댓글 내용 입력 후 Enter 혹은 버튼 클릭시 댓글 추가 기능
- [추가구현 Mission5] ID,PW 유효성 검사 구현 (id에 @포함 및 pw 5자 이상)

sunnyfterrain committed:
Add: 댓글 좋아요/삭제 기능 추가
- [Mission6] 댓글 좋아요 및 삭제 기능 추가

sunnyfterrain committed:
Add: 아이디 검색기능, 프로필 메뉴 기능, 반응형 구현
- [Mission7] 아이디 검색기능 추가, 영어, 한글 검색기능, 대소문 구분x
- [Mission8] nav 프로필 사진 클릭시 메뉴박스 생성, 바깥화면 클릭시 닫힘
- [Mission9] 반응형 구현 브라우저 1000px 이하 시 우측 메뉴 사라짐

sunnyfterrain committed:
Fix: 프로필 버튼 클릭 이벤트 수정
- 프로필 모달 제대로 뜨지 않고 원하지 않게 창이 닫혀 수정

sunnyfterrain committed:
Modify: fetch 주소 수정 및 프로필 모달창 수정
- fetch를 못불러오는것에 대한 수정
- 프로필 모달창 외부 클릭하면 닫힘 기능 재구현
